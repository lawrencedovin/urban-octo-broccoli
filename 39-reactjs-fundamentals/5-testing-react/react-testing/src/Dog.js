import React from 'react';

const Dog = (({name, isAdopted}) => {
    return (
        <div className="Dog">
            <h1>{name}</h1>
            {isAdopted ? 
            (<p>{name} has been adopted!</p>) 
            : 
            (<button>Adopt me!</button>)
            }
        </div>
    )
});

export default Dog;