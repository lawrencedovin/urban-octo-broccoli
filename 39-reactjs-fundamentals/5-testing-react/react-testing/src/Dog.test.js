import React from 'react';
import { render } from '@testing-library/react';
import Dog from './Dog';

it('should render w/o crashing', () => {
  render(<Dog name='Pancake' isAdopted={true} />);
});

it('should match snapshot with isAdopted to true', () => {
    const {asFragment} = render(<Dog name='Pancake' isAdopted={true} />);
    expect(asFragment()).toMatchSnapshot();
})

it('should match snapshot with isAdopted to false', () => {
    const {asFragment} = render(<Dog name='Pancake' isAdopted={false} />);
    expect(asFragment()).toMatchSnapshot();
})