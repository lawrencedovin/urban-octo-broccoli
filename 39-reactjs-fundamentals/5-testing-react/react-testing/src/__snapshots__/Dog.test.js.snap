// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Pancake
    </h1>
    <p>
      Pancake has been adopted!
    </p>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot with isAdopted to false 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Pancake
    </h1>
    <button>
      Adopt me!
    </button>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot with isAdopted to true 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Pancake
    </h1>
    <p>
      Pancake has been adopted!
    </p>
  </div>
</DocumentFragment>
`;
